R/xdfutils.R

```diff
@@ -23,18 +23,22 @@ getXdfLevels <- function(formula, xdf) {
 #'  whether data is XDF and the path of the data
 #' @import rjson
 #' @export
-getXdfProperties <- function(input_name) {
-  is_XDF <- FALSE
-  meta_data <- AlteryxRDataX::read.AlteryxMetaInfo(input_name)
-  the_source <- as.character(meta_data$Source)
-  if (all(substr(the_source, 3, 9) == "Context")) {
-    context_list <- rjson::fromJSON(the_source[1])
-    if (context_list$Context == "XDF") {
-      is_XDF <- TRUE
-      xdf_path <- context_list$File.Loc
-      list(is_XDF = is_XDF, xdf_path = xdf_path)
-    } else {
-      stop.Alteryx2("At this time only XDF scaling is supported.")
+getXdfProperties <- function(input_name, default = NULL) {
+  if(inAlteryx()){
+    default
+  } else {
+    is_XDF <- FALSE
+    meta_data <- AlteryxRDataX::read.AlteryxMetaInfo(input_name)
+    the_source <- as.character(meta_data$Source)
+    if (all(substr(the_source, 3, 9) == "Context")) {
+      context_list <- rjson::fromJSON(the_source[1])
+      if (context_list$Context == "XDF") {
+        is_XDF <- TRUE
+        xdf_path <- context_list$File.Loc
+        list(is_XDF = is_XDF, xdf_path = xdf_path)
+      } else {
+        stop.Alteryx2("At this time only XDF scaling is supported.")
+      }
```

Comment:
You need to return `list(is_XDF = FALSE, xdf_path = NULL)` if the meta data does not have any `context`. Otherwise, this function returns a NULL and as a result, `XDFInfo$is_XDF` is not defined downstream.